components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -59,10 +60,18 @@ class DatabaseLoginsStorage(private val dbPath: String) : AutoCloseable, LoginsS
             if (!isLocked()) {
                 throw MismatchedLockException("Unlock called when we are already unlocked")
             }
-            raw.set(PasswordSyncAdapter.INSTANCE.sync15_passwords_state_new(
-                    dbPath,
-                    encryptionKey,
-                    it))
+            val timer = LoginsStoreMetrics.dbUnlockTime.start()
+            try {
+                raw.set(PasswordSyncAdapter.INSTANCE.sync15_passwords_state_new(
+                        dbPath,
+                        encryptionKey,
+                        it))
+            } finally {
+                LoginsStoreMetrics.dbUnlockTime.stopAndAccumulate(timer)
+            }
```

Comment:
I wonder if Glean would consider adding some sort of `LoginsStoreMetrics.dbUnlockTime.execute` helper to encapsulate this start/stop pair...

Reply:
That's a good API extension :) Would you kindly consider filing [a bug](https://bugzilla.mozilla.org/enter_bug.cgi?product=Data%20Platform%20and%20Tools&component=Glean%3A%20SDK)? Please note you could also add this to your repo by using Kotlin extension functions:

```kotlin
        fun TimingDistributionMetricType.execute(funcToMeasure: () -> Unit) {
            val timerId = this.start()
            try {
                funcToMeasure()
            } finally {
                this.stopAndAccumulate(timerId)
            }
        }
```